haskell-src/Concordium/Types/Queries/Tokens.hs

```diff
@@ -77,14 +79,47 @@ data TokenState = TokenState
       -- | The governance account for the token.
       tsIssuer :: !AccountAddress,
       -- | The number of decimals in the token representation.
-      tsDecimals :: !Word8,
+      tsDecimals :: !Word32,
       -- | The total available token supply.
       tsTotalSupply :: !TokenAmount,
       -- | CBOR-encoded module-specific state.
       tsModuleState :: !BS.ByteString
     }
     deriving (Eq, Show)
 
+-- Nice-to-have: Use a wrapper for dispalying the decoded CBOR representation instead of displaying the bytes as a hex string.
+newtype HexByteString = HexByteString {_unHex :: BS.ByteString}
+
+instance ToJSON HexByteString where
+    toJSON (HexByteString bs) =
+        String (Text.decodeUtf8 (BS16.encode bs))
+
+instance FromJSON HexByteString where
+    parseJSON = AE.withText "HexByteString" $ \txt ->
+        case BS16.decode (Text.encodeUtf8 txt) of
+            Right bs -> pure $ HexByteString bs
+            _ -> fail "Invalid Base16 encoding"
```

Comment:
Move this to `ByteStringHelpers` with the rest of these. I'm not sure it's great to have both `ByteStringHex` and `HexByteString`.

Reply:
Moved it and renamed the already existing type`ByteHexString` to `ShortByteHexString`.